hopp/offshore/floating_platform.py

```diff
@@ -0,0 +1,345 @@
+            "depth" : "int | float",
+        }, 
+
+        "equipment": {
+            "tech_required_area" : "float", 
+            "tech_combined_mass" : "float",
+            "install_duration": "days (optional, default: 14)",
+        },
+
+        "oss_install_vessel" : "str | dict",
+    }
+
+    # Need to initialize arguments and weather files 
+    def __init__(self, config, weather=None, **kwargs):
+        
+        super().__init__(weather, **kwargs)
+
+        config = self.initialize_library(config, **kwargs)
+        self.config = self.validate_config(config)
+
+        self.initialize_port()
+        self.setup_simulation(**kwargs)
+
+    # Setup simulation seems to be the install phase's equivalent run() module
+    def setup_simulation(self, **kwargs):
+
+        #print("Floating Platform Install setup_sim() is working!!!")
+
+        self.distance = self.config['site']['distance']
+        self.depth = self.config['site']['depth']
+        self.mass = self.config['equipment']['tech_combined_mass']
+        self.area = self.config['equipment']['tech_required_area']
+
+        _platform = self.config.get('equipment', {})
+        design_cost = _platform.get('topside_design_cost', 4.5e6)   # USD
+        fab_cost = _platform.get('fabrication_cost_rate', 14500.)   # USD/t
+        steel_cost = _platform.get('substructure_steel_cost', 3000) # USD/t
+        #NEED Updated version for floating
+        install_duration = _platform.get("install_duration", 14)    # days
+        
+        # Initialize vessel 
+        vessel_specs = self.config.get("oss_install_vessel", None)
+        name = vessel_specs.get("name","Offshore Substation Install Vessel")
+
+        vessel = Vessel(name, vessel_specs)
+        self.env.register(vessel)
+
+        vessel.initialize()
+        self.install_vessel = vessel
+        
+        # Add in the mass of the substructure to total mass (may or may not impact the final install cost)
+        _, substructure_mass = calc_substructure_mass_and_cost(self.mass, self.area, 
+                        self.depth, fab_cost, design_cost, steel_cost
+                        )
+
+        total_mass = self.mass + substructure_mass  # t
+
+         # Call the install_platform function
+        self.install_capex = install_platform(total_mass, self.area, self.distance, \
+                                                   install_duration, self.install_vessel)
+
+    # An install object needs to have attribute system_capex, installation_capex, and detailed output
+    @property
+    def system_capex(self):
+
+        return {}
+
+    @property 
+    def installation_capex(self):
+        
+        return self.install_capex
+
+    @property
+    def detailed_output(self):
+
+        return {}
+
+# Define individual calculations and functions to use outside or with ORBIT
+def calc_substructure_mass_and_cost(mass, area, depth, fab_cost=14500., design_cost=4.5e6, sub_cost=3000, pile_cost=0):
```

Comment:
- `fab_cost` seems to be used as a rate. The name should reflect that.
- please add doc strings to the functions specifying what each input is, where the default values come from, what the units are, etc
-